internal/v5/archive_test.go

```diff
@@ -1729,3 +1756,125 @@ func (s *ArchiveSearchSuite) TestGetSearchUpdate(c *gc.C) {
 		stats.CheckSearchTotalDownloads(c, s.store, &url.URL, 1)
 	}
 }
+
+type ArchiveSuiteWithTerms struct {
+	commonArchiveSuite
+}
+
+var _ = gc.Suite(&ArchiveSuiteWithTerms{})
+
+func (s *ArchiveSuiteWithTerms) SetUpSuite(c *gc.C) {
+	s.commonSuite.SetUpSuite(c)
+	s.enableTerms = true
+	s.enableIdentity = true
+}
+
+func (s *ArchiveSuiteWithTerms) SetUpTest(c *gc.C) {
+	s.commonSuite.SetUpTest(c)
+	s.discharge = dischargeForUser("bob")
+}
+
+func (s *ArchiveSuiteWithTerms) TestGetUserHasAgreedToTermsAndConditions(c *gc.C) {
+	s.dischargeTerms = func(s1, s2 string) ([]checkers.Caveat, error) {
```

Comment:
And also let's leave something around to ensure us that the terms discharge checker *is* in fact being called in this case.